osquery/filesystem/filesystem.cpp

```diff
@@ -147,6 +148,120 @@ Status listFilesInDirectory(const boost::filesystem::path& path,
   }
 }
 
+Status listDirectoriesInDirectory(const boost::filesystem::path& path,
+                            std::vector<std::string>& results) {
+  try {
+    if (!boost::filesystem::exists(path)) {
```

Comment:
https://github.com/facebook/osquery/blob/master/osquery/filesystem/filesystem.cpp#L110

And you can move this out of a try.